File: wagerproof-ios-native/WagerproofKit/Sources/WagerproofStores/CFBGameSheetStore.swift
import Foundation
import Observation
import WagerproofModels

/// Drives the CFB game detail bottom sheet. Mirrors the RN
/// `CFBGameSheetContext` (provider + `useCFBGameSheet` hook). See
/// `NFLGameSheetStore` — identical shape, different model.
@Observable
@MainActor
public final class CFBGameSheetStore {
    public var selectedGame: CFBPrediction?

    public init() {}

    public func openGameSheet(_ game: CFBPrediction) {
        selectedGame = game
    }

    public func closeGameSheet() {
        selectedGame = nil
    }
}

File: wagerproof-android-native/core/stores/src/main/kotlin/com/wagerproof/core/stores/CFBGameSheetStore.kt
package com.wagerproof.core.stores

import androidx.compose.runtime.Stable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import com.wagerproof.core.models.CFBPrediction

/**
 * Drives the CFB game detail bottom sheet. See [NFLGameSheetStore] — identical
 * shape, different model.
 */
@Stable
class CFBGameSheetStore {
    var selectedGame by mutableStateOf<CFBPrediction?>(null)

    fun openGameSheet(game: CFBPrediction) {
        selectedGame = game
    }

    fun closeGameSheet() {
        selectedGame = null
    }
}
